pkg/kademlia/kademlia_planet_test.go

```diff
@@ -131,35 +131,45 @@ func TestBootstrapBackoffReconnect(t *testing.T) {
 	go func() {
 		// This starts the unreliable proxy server and sets it to
 		// drop connections for the first 500 milliseconds that it's up.
-		err := proxy.start(done, droppedConnInterval)
+		err := proxy.run(done, droppedConnInterval)
+		require.NoError(t, err)
+	}()
+	defer func() {
+		err := proxy.close()
```

Comment:
This doesn't wait for the goroutine to be finished.

Reply:
@egonelbre how should I wait for the goroutine to be finished?

Reply:
it's not working when I try to add `done <- true` in this defer.

Reply:
If something internally fails in the loop, it isn't propagated to t.Fatal / require.NoError, but instead it's only being logged at the moment.

Reply:
To be clear. No goroutine should be running once the test finishes. All errors must be properly handled, which means either explicitly ignored (e.g. context.Canceled) or checked with require.Erorr.